### ISSUE TITLE
New icontasklist "breaks things"

### ISSUE THREAD

**Issue by ikeydoherty** [solus-project/budgie-desktop#1094]
Preempting this with a heads up so we can discuss it:

 - [ ] Probably breaks 3.18 theming with `:drop` selector -  need verification
 - [ ] Will certainly cause issues with Adapta theme - discuss
 - [ ] Grouping is not configurable, it's the default and only behaviour .. Make it optional
 - [ ] Advanced window actions now gone from icontasklist right-click menu - do we want them?
 - [ ] Window dots unbounded, i.e. we don't cap the number of dots we draw (should likely be something like 5)
 - [ ] Intermittent bug with duplicate window buttons for grouping

Theming:

Right now the tasklist itself renders the "dots", which might cause issues for themers. So I'm poking @tista500 to discuss this, whether it should be CSS domain only (i.e. `.one-dot .two-dot`) or we could extend the new cairo code to compensate and offer more flexibility.

**Comment by ikeydoherty** [comment]
Once this is addressed we can clean up the tree and set about getting ready for 10.4

**Comment by Vistaus** [comment]
Add:

-WPS Icon too big for panel

**Comment by fossfreedom** [comment]
aye - on GTK+3.18 dropping the :drop selector in _workspaces.scss allows the theme to load correctly.

**Comment by tista500** [comment]
Fixed conflicts with your dot-based indicators: https://github.com/adapta-project/adapta-gtk-theme/commit/61b08f667106e0c8f07e7625405c41364f930694

**Comment by tista500** [comment]
Fixed hovering effects: https://github.com/adapta-project/adapta-gtk-theme/commit/a610a5d846f488f08ae31e6a8863e35ada93a216

> whether it should be CSS domain only (i.e. .one-dot .two-dot) or we could extend the new cairo code to compensate and offer more flexibility.

I'm OK with cairo codes, but have some questions:
  1. `.running` style-class was dropped?
    We could not set visual contrasts to 'launched' apps icons against !launched apps.
  2. launched and !focused apps's dots might be too translucent?
    Visual contrast 'focused' vs. '!focused' are fine, but on some case (Dynamic mode + bright wallpapers) those might cause missing dots, 

Just my thoughts. ;)

**Comment by ikeydoherty** [comment]
Run into a bug here @cybre 

![screenshot from 2017-08-08 10-41-52](https://user-images.githubusercontent.com/164489/29066160-4ab2a5e0-7c26-11e7-8533-cf4ed0a1ee4e.png)

**Comment by cybre** [comment]
I will be available to discuss this tonight.

**Comment by ikeydoherty** [comment]
@cybre again - no rush, I know you've got shit going on. Just making you aware of it, and it'll likely be me making the fixes, so no panics on that front. :)

**Comment by ikeydoherty** [comment]
Another issue, with two terminals grouped, there isn't a concept of a primary window. So both windows will be minimised/restored together.

**Comment by cybre** [comment]
This is by design.
```
If all windows are visible (unminimized):
                        If there is one active window:
                     *     minimize all windows
                     *   else:
                     *     activate all windows
                     * else:
                     *   activate all minimized windows
                     */
```

**Comment by cybre** [comment]
This is by design.
```
 If all windows are visible (unminimized):
   If there is one active window:
     minimize all windows
   else:
     activate all windows
 else:
   activate all minimized windows
```

This is what I've seen in most docks I've used and this is what feels right to me.

**Comment by ikeydoherty** [comment]
Right but those are docks, Budgie is primarily a panel that has a dock *mode*. So if I click the tray button or minimise my terminals, they go back to the tasklist. If I click it, then both of them show up .. instead of being able to just bring **one** back up (which is what Windows does iirc)

**Comment by ikeydoherty** [comment]
@tista500 apologies my good sir - I've reverted the icontasklist changes until 10.4.1 - we're gonna roll with the current tasklist for now, to unblock release. 10.4.1 will feature the more mature version of the tasklist, but this now gives us time to properly plan and implement it, and ensure it enables **you** as a theme artist.

**Comment by tista500** [comment]
@ikeydoherty,

Ah, OK. At least I like those changes with cute animations, apps grouping and responsive dot implementations... :)

Best Regards.

**Comment by ikeydoherty** [comment]
Yeah +1 on the cuteness :D - we'll have this in .1 though so its not *too* far away - just gonna make sure it goes in solid :]

**Comment by JoshStrobl** [comment]
Closing issue as:

1. We no longer support 3.18, thus initial issue moot.
2. Issues no longer exist and upstream is dead anyways.
3. Grouping is configurable.
4. Will be in-progress with #1428 and prefer to track there.
